src/Channels/ReadableBuffer.cs

```diff
@@ -572,6 +573,115 @@ internal static int FindFirstEqualByteSlow(ref Vector<byte> byteEquals)
             throw new InvalidOperationException();
         }
 
+
+        /// <summary>
+        /// Reads a structure of type T out of a buffer of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public T ReadBigEndian<[Primitive]T>() where T : struct
```

Comment:
I wonder if these should be on the ReadableBuffer itself or if we need BinaryReader/Write for different endianess